models/gyroscope/ethereum/gyroscope_ethereum_trades.sql

```diff
@@ -0,0 +1,35 @@
+{{ config(
+    schema = 'gyroscope_ethereum',
+    alias = 'trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook = '{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "gyroscope",
+                                \'["fmarrr"]\') }}'
+    )
+}}
+
+with E_CLPs AS (
+    SELECT 
+        x.pool
+        , y.min_block_time
+        FROM {{ source('gyroscope_ethereum','GyroECLPPoolFactory_evt_PoolCreated') }} x
+        LEFT JOIN (
+            SELECT min(evt_block_time) AS min_block_time 
+            FROM {{ source('gyroscope_ethereum','GyroECLPPoolFactory_evt_PoolCreated') }} 
+        ) y
+        on x.pool is not null
+)
+
+SELECT
+ *,
+ CASE
+  when project_contract_address IN (SELECT pool FROM E_CLPs) THEN 'gyroscope' end as project
```

Comment:
Since you inner join the `E_CLPs` table you do not need this CASE statement to check if the pool is in the E_CLPs table, as the inner join will only return E_CLP pools.

You could simply just do: `'gyroscope' AS project`